rpc/src/eth_api.rs

```diff
@@ -0,0 +1,240 @@
+use jsonrpsee::{core::RpcResult as Result, proc_macros::rpc};
+use reth_primitives::{
+    rpc::transaction::eip2930::AccessListWithGasUsed, Address, BlockId, BlockNumberOrTag, Bytes,
+    H256, H64, U256, U64,
+};
+use reth_rpc_types::{
+    CallRequest, EIP1186AccountProofResponse, FeeHistory, Index, RichBlock, SyncStatus,
+    Transaction as EthTransaction, TransactionReceipt, TransactionRequest, Work,
+};
+
+#[rpc(server)]
+#[async_trait]
+pub trait EthApi {
```

Comment:
this trait doesn't exist yet in reth?

Reply:
there's probably a way but I don't know it yet, let's keep this in mind while (re-)reading the rust book